### PR TITLE
Implement unit test for the hook (useBuildingEvents) 

### PR DESCRIPTION
**Pull request by yas176**
This pull request should close task #208     

### What's tested    

**1-  Location Parsing (4 tests)**  

- Valid location formats (space and dash separators)  
- Case-insensitive building code matching  
- Invalid/empty location handling  

**2-  Event Filtering (4 tests)**  

- Filter events by exact building code match  
- Return empty results when no events match  
- Handle empty buildingId parameter  
- Correctly populate buildingCode and roomNumber  

**4- Today's Events (3 tests)**  

- Filter only today's events correctly  
- Sort today's events by start time  
- Return empty array when no today's events exist  

**5- Next Event (3 tests)**  

- Identify the next upcoming event correctly  
- Return null when no upcoming events exist  
- Return null when events array is empty  

**6- State Management (2 tests)**   

- Propagate loading state from useGoogleCalendar  
- Propagate error state from useGoogleCalendar  

**7- Edge Cases (3 tests)**  

- Handle events with date-only format (no dateTime)  
- Verify all required BuildingEvent fields are present  
- Ensure start/end are converted to Date objects  

### Test Coverage  
- Total Tests: 19  
- All Passing.

### CODE REVIEW

File: frontend/conuwalks/src/_tests_/useBuildingEvents.test.ts [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+  },
+}));
+
+// Then import and mock useGoogleCalendar
+jest.mock("@/src/hooks/useGoogleCalendar", () => ({
+  useGoogleCalendar: jest.fn(),
+}));
+
+import { renderHook, waitFor } from "@testing-library/react-native";
+import { useBuildingEvents } from "@/src/hooks/useBuildingEvents";
```
[the remark this answers is not in the document]
Comment (yas176) [reply frontend/conuwalks/src/_tests_/useBuildingEvents.test.ts]: You can pull from elif branch elif/api to see changes she made. this test is implemented to test the hook she created (useBuildingEvents). Since it caused confusion, I will wait for her to merge to dev then I will do the test and push again.

[the remark this answers is not in the document]
Comment (rwatoun) [reply frontend/conuwalks/src/_tests_/useBuildingEvents.test.ts]: I see perfect thank you for clarifying!
